crates/ruff_python_formatter/tests/snapshots/format@statement__long_type_annotations.py.snap

```diff
@@ -0,0 +1,451 @@
+)
+
+leading_open_parentheses_with_value: (  #  Leading comment
+    None | dataset.ImageDataset
+) = None
+
+trailing: (
+    None | dataset.ImageDataset  # trailing comment
+)
+
+trailing_with_value: (
+    None | dataset.ImageDataset  # trailing comment
+) = None
+
+trailing_own_line: (
+    None | dataset.ImageDataset
+    #  trailing own line
+)
+
+trailing_own_line_with_value: (
+    None | dataset.ImageDataset
+    #  trailing own line
+) = None
+
+nested_comment: None | [
+    # a list of strings
+    str
+] = None
+```
+
+
+## Preview changes
+```diff
+--- Stable
++++ Preview
+@@ -1,8 +1,18 @@
+-x1: A[b] | EventHandler | EventSpec | list[
+-    EventHandler | EventSpec
+-] | Other | More | AndMore | None = None
++x1: (
++    A[b]
++    | EventHandler
++    | EventSpec
++    | list[EventHandler | EventSpec]
++    | Other
++    | More
++    | AndMore
++    | None
++) = None
+ 
+-x2: "VeryLongClassNameWithAwkwardGenericSubtype[int] |" "VeryLongClassNameWithAwkwardGenericSubtype[str]"
++x2: (
++    "VeryLongClassNameWithAwkwardGenericSubtype[int] |"
++    "VeryLongClassNameWithAwkwardGenericSubtype[str]"
++)
+ 
+ x6: VeryLongClassNameWithAwkwardGenericSubtype[
+     integeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeer,
+@@ -48,12 +58,16 @@
+     datasets.VideoDataset,
+ ] = None
+ 
+-x12: None | [
+-    datasets.ImageDataset,
+-    datasets.TabularDataset,
+-    datasets.TextDataset,
+-    datasets.VideoDataset,
+-] | Other = None
++x12: (
++    None
++    | [
++        datasets.ImageDataset,
++        datasets.TabularDataset,
++        datasets.TextDataset,
++        datasets.VideoDataset,
++    ]
++    | Other
++) = None
+ 
+ 
+ x13: [
+@@ -75,27 +89,34 @@
+     datasets.VideoDataset,
+ ] = None
+ 
+-x15: [
+-    datasets.ImageDataset,
+-    datasets.TabularDataset,
+-    datasets.TextDataset,
+-    datasets.VideoDataset,
+-] | [
+-    datasets.ImageDataset,
+-    datasets.TabularDataset,
+-    datasets.TextDataset,
+-    datasets.VideoDataset,
+-] | Other = None
++x15: (
++    [
++        datasets.ImageDataset,
++        datasets.TabularDataset,
++        datasets.TextDataset,
++        datasets.VideoDataset,
++    ]
++    | [
++        datasets.ImageDataset,
++        datasets.TabularDataset,
++        datasets.TextDataset,
++        datasets.VideoDataset,
++    ]
++    | Other
++) = None
+ 
+-x16: None | Literal[
+-    "split",
+-    "a bit longer",
+-    "records",
+-    "index",
+-    "table",
+-    "columns",
+-    "values",
+-] = None
++x16: (
++    None
++    | Literal[
++        "split",
++        "a bit longer",
++        "records",
++        "index",
++        "table",
++        "columns",
++        "values",
++    ]
++) = None
+ 
+ x17: None | [
+     datasets.ImageDataset,
+@@ -106,9 +127,13 @@
+ 
+ 
+ class Test:
+-    safe_age: Decimal  #  the user's age, used to determine if it's safe for them to use ruff
++    safe_age: (
++        Decimal  #  the user's age, used to determine if it's safe for them to use ruff
```

Comment:
I'm not a huge fan of this change but it is consistent to how we would format the same assignment when `Decimal`is the assigned value

Input 
```python
safe_age = Decimal #  the user's age....
```